### ISSUE TITLE
QInput/QSelect:  Hide label slot when using null label

### ISSUE THREAD

**Issue by charliegilmanuk** [quasarframework/quasar#7837]
I am migrating a large project from Vuetify to Quasar and so far has been well worth it, improving performance and cutting down massively on 3rd party libraries, however I am missing the ability to hide the label slot for inputs, as part of the design requires this.

This was done in Vuetify through use of the [`single-line` prop](https://vuetifyjs.com/en/components/text-fields/#single-line).  Omitting the label prop has the desired effect, _however_ I am using extended components which set a default label, and setting the label to `null` does not do the same thing.

## Solution:

Setting the label to null or using a new `hide-label` prop should:
- Show the placeholder (if available), current functionality makes the placeholder hidden until the input has focus or a value weirdly
- Hide the padding above the input where the label would go

#### AND/OR

Introduce `single-line` prop that would:
- Basically use the label as the placeholder attribute (see Vuetify link above)
- Hide the padding above the input where the label would go

## Comparison

Vuetify | Quasar
--------|-------
![Vuetify example](https://i.imgur.com/ReP1wXL.jpg) | ![Quasar implementation](https://i.imgur.com/OQapBoE.jpg)

**Notes**:
The two date fields are non-extended components and are working as intended as their label prop is completely omitted.  The others are doing all sorts of strange things to note:
- No placeholders until they either have focus or a value
- Placeholder still showing even when the field has a value
- Top padding is still included even though the label prop is set to null

**Comment by rstoenescu** [comment]
Hi and welcome to Quasar.
You can specify `void 0` (or `undefined`) and it'll treat as if the property is not there. No need to add extra props in components. This is a general rule of thumb for all components.

```
:label="someCondition ? 'some label' : void 0"
```

**Comment by charliegilmanuk** [comment]
> Hi and welcome to Quasar.
> You can specify `void 0` (or `undefined`) and it'll treat as if the property is not there. No need to add extra props in components. This is a general rule of thumb for all components.
> 
> ```
> :label="someCondition ? 'some label' : void 0"
> ```

Wasn't aware of that functionality, good to know but unfortunately doesn't solve the problem, the extended component rightly assumes the `label` prop is absent and uses the default value set by said component.

I've managed to get the labels hidden and the placeholder showing correctly using this:

```javascript
if (this.$attrs.label === null) {
    delete this.$attrs.label;
}
```

However the field is still much higher than it should be.  Written some hacky CSS to fix it for now, I'll see if I can whip up a codepen as I understand my use case is getting a little confusing!

**Comment by charliegilmanuk** [comment]
And weirdly the placeholder is still shown when the inputs have a value, this is where I'm at now with the above:

[Imgur](https://i.imgur.com/bVCgjwD.png)

**Comment by rstoenescu** [comment]
Can you post your component here pls? Want to fully understand your usage. If it's too heavy, attach it as a file. Otherwise please contact me on Discord (https://chat.quasar.dev) and share it with me on private. Thanks and we'll get to the bottom of this.

**Comment by charliegilmanuk** [comment]
Wasn't possible to describe in a single component, I created a quick project demonstrating it here:  https://www.dropbox.com/s/xazuzbx7agegxvk/quasar-input-labels.zip?dl=1

I did have some custom CSS from before the migration that was affecting the height slightly which I have now removed, however the content still isn't vertically aligned center.  I realise this part is probably down to me using custom sized avatars in the `selected-item` slot so I can use CSS to fix that no problem.

The main issue is the placeholder text now, as annotated below:

![Imgur](https://i.imgur.com/SusEvGs.jpg)

**Comment by rstoenescu** [comment]
Thanks for supplying a reproduction repo.
I've investigated a bit, and "label" is still assigned a value of null instead of void 0/undefined.

**Comment by rstoenescu** [comment]
McQuoteStatusSelector > mounted > console.log({ ...this.$attrs, ...this.$props }) -> label: null

**Comment by rstoenescu** [comment]
You can override that by adding `:label="void 0"` or making a computed property and assigning it to the label prop, besides the v-bind="...".

**Comment by rstoenescu** [comment]
Please reopen or continue commenting if I missed anything.

**Comment by mpont91** [comment]
Hi @rstoenescu 

I have this issue on a custom component that combines a text field with a select.

The solutions proposed don't work in this case, I reproduced with a codepen:

https://codepen.io/slayerbleast/pen/XWqrrqL

I tried with:

```
:label="void 0" 
```

and 

```
:label="undefined"
```

In both cases the select has reserved the space for the label.

**Comment by EvandroLucas** [comment]
I'm having the same problem right now. Any updates on this? Label is null, undefined, or void 0, but the label slot is still there occupying space.

**Comment by rstoenescu** [comment]
I'm looking at https://codepen.io/slayerbleast/pen/XWqrrqL
It would be best not to put QSelect inside QInput. Just insert it as a sibling of QInput instead, otherwise props might get mixed up, as it happens there with the label from QInput.
